File: SampleProjects/2.1/MixiNavigationSample2/MixiNavigationSample2/ViewController.m
//
//  ViewController.m
//  MixiNavigationSample2
//
//  Created by yuichi.takeda on 2/4/15.
//  Copyright (c) 2015 mixi, Inc. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = [NSString stringWithFormat:@"hoge%ld", self.navigationController.viewControllers.count];
}

- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    UIViewController *viewController = [self.storyboard instantiateViewControllerWithIdentifier:@"MixiViewController"];
    UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:viewController];
    [self presentViewController:navigationController animated:YES completion:nil];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swift/after/day2/1.1/InfiniteNavigationSample/InfiniteNavigationSample/ViewController.swift
//
//  ViewController.swift
//  InfiniteNavigationSample
//
//  Created by 鈴木 大貴 on 2016/10/23.
//  Copyright © 2016年 szk-atmosphere. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        let string: String
        if let count = navigationController?.viewControllers.count {
            string = "\(count)"
        } else {
            string = ""
        }
        title = string
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func pushButtonTapped(_ sender: UIButton) {
        guard let viewControlelr = storyboard?.instantiateViewController(withIdentifier: "ViewController") as? ViewController else {
            return
        }
        navigationController?.pushViewController(viewControlelr, animated: true)
    }
}

